week3/xor/jasonkuhrt/Main.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE OverloadedStrings #-}
+{- README
+
+-- XOR Encryption
+
+-- Spec
+
+Write a program that encrypts a file by XORing it with a key. For instance:
+
+     Key  "XYZ"
+Contents  "abracadabra"
+     XOR   XYZXYZXYZXY
+       =   9;(9:;<88*8
+
+* It should accept the file name and the key as command-line arguments
+* Use bytestring to read the file
+* Encode the key as UTF8 (with encodeUtf8)
+* Overwrite the file with encryped variant
+
+-- Notes
+
+Due to the way XOR works, encrypting an already encrypted file will decrypt
+it. Learn more about XOR Ciphers at en.wikipedia.org/wiki/xor_cipher.
+-}
+
+
+
+module Main where
+
+import qualified Data.Bits as Bits
+import qualified System.Environment as Env
+import Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as B
```

Comment:
The usual name is `BSL` or `BL` for lazy bytestrings (and `T` or `TL` for text, for that matter.)